lib/dukpt/decrypter.rb

```diff
@@ -24,9 +24,9 @@ def decrypt_pin(cryptogram, ksn, pan)
       pan &&= pan.downcase.chomp('f')
       decrypted_block = decrypt_pin_block(cryptogram, ksn).unpack("H*").first
       block_format = decrypted_block[0]
+      coded_pan = "0000"+pan[-13..-2]
+      coded_pin = (decrypted_block.to_i(16) ^ coded_pan.to_i(16)).to_s(16).rjust(16, '0')
       if block_format == "0"
-        coded_pan = "0000"+pan[-13..-2]
-        coded_pin = (decrypted_block.to_i(16) ^ coded_pan.to_i(16)).to_s(16).rjust(16, '0')
         pin_count = coded_pin[1].to_i
         coded_pin[2,pin_count]
       elsif block_format == "1"
```

Comment:
If I'm remembering this right, block format 1 does not require access to the PAN. Referencing `coded_pin` in here instead of `decrypted_block` might be the bug instead. Any chance you have block 1 test fixture?